### PR TITLE
Move section about ltac1 defects to ltac.rst instead of ltac2.rst

### PR DESCRIPTION
**Pull request by SkySkimmer**
The ltac2 page should be mostly about ltac2, not ltac1.

### CODE REVIEW

File: doc/sphinx/proof-engine/ltac.rst [hunk 1]
```diff
@@ -39,6 +39,48 @@ since the paper was written.)
...
+
+Following the need of users who are developing huge projects relying
+critically on Ltac, we believe that we should offer a proper modern language
+that features at least the following:
+
+- at least informal, predictable semantics
+- a type system
+- standard programming facilities (e.g., datatypes)
+
+This new language, called Ltac2, is described in this chapter. It is still
```
Comment (proux01) [inline doc/sphinx/proof-engine/ltac.rst:65]: Isn't that wrong now?

Comment (SkySkimmer) [reply doc/sphinx/proof-engine/ltac.rst]: changed

**Review by proux01 — APPROVED** [review]
Will merge when CI is happy.

**Comment by proux01** [comment]
Maybe this could also mention other alternatives like elpi but that can be done in another PR.

File: doc/sphinx/proofs/creating-tactics/index.rst
Comment (SkySkimmer) [inline doc/sphinx/proofs/creating-tactics/index.rst:1]: elpi is mentioned in this file

**Comment by Zimmi48** [comment]
> Maybe this could also mention other alternatives like elpi but that can be done in another PR.

The note at the top of the chapter (already before this PR) mentions that there are other alternatives than Ltac2 and directs to the Rosetta Stone of metaprogramming to learn about these.

**Comment by proux01** [comment]
@coqbot merge now

**Comment by Zimmi48** [comment]
Currently, the note at the top of the Ltac1 chapter says: "Reports about hindrances in using Ltac2 for writing automation are welcome as issues on the [Coq bug tracker](https://github.com/coq/coq/issues) or as discussions on the [Ltac2 Zulip stream](https://coq.zulipchat.com/#narrow/stream/278935-Ltac2)."

What about repeating the same piece of advice at the top of the Ltac2 chapter?

I also wonder if the first sentence of the Ltac2 chapter should keep listing "Mtac or Rtac".

**Pull request merged**